src/muse/demand_share.py

```diff
@@ -630,37 +638,19 @@ def new_and_retro_demands(
         .groupby("region")
         .sum("asset")
     )
-    # existing asset should not execute beyond demand
+
+    # Existing asset should not execute beyond demand
     service = minimum(
         service, smarket.consumption.sel(year=current_year + forecast, drop=True)
     )
+
+    # Leftover demand that cannot be serviced by existing assets or "new" agents
     retro_demand = (
         smarket.consumption.sel(year=current_year + forecast, drop=True)
         - new_demand
         - service
     ).clip(min=0)
-
     if "year" in retro_demand.dims:
         retro_demand = retro_demand.squeeze("year")
 
     return xr.Dataset({"new": new_demand, "retrofit": retro_demand})
-
-
-def new_demand(
```

Comment:
I remember to have coded this when removing the retrofit agents, and I guess I did it for a reason to be used somewhere, but it is true that it doesn't seem to be used at all. Well, bye bye function!